Python/compile.c

```diff
@@ -184,31 +185,71 @@ static struct jump_target_label_ NO_LABEL = {-1};
         return 0; \
     }
 
+enum oparg_type {
+    OPARG_TYPE_UNUSED,
+    OPARG_TYPE_EXPLICIT, /* int value */
+    OPARG_TYPE_CONST,    /* index of const */
+    OPARG_TYPE_NAME,     /* index of name */
+    OPARG_TYPE_TMP,      /* index of tmp */
+};
+
+typedef struct oparg_ {
+    enum oparg_type type;
+    int value; /* logical value set by codegen */
+    int final; /* actual reg value, resolved in assembly */
+} oparg_t;
+
+#define UNUSED_OPARG      ((const oparg_t){.value=(0), .type=OPARG_TYPE_UNUSED})
+#define EXPLICIT_OPARG(V) ((const oparg_t){.value=(V), .type=OPARG_TYPE_EXPLICIT})
+#define CONST_OPARG(V)    ((const oparg_t){.value=(V), .type=OPARG_TYPE_CONST})
+#define NAME_OPARG(V)     ((const oparg_t){.value=(V), .type=OPARG_TYPE_NAME})
+#define TMP_OPARG(V)      ((const oparg_t){.value=(V), .type=OPARG_TYPE_TMP})
+
+#define IS_UNUSED(OPARG) ((OPARG).type == OPARG_TYPE_UNUSED)
+#define SAME_REGISTER(R1, R2) (((R1).type == (R2).type) && ((R1).value == (R2).value))
+
+typedef struct instr_opargs_ {
+    oparg_t arg1;
```

Comment:
Once we have variable-length instructions, arg2, arg3 should be added here. Then search this file for arg1 and update all those places to work with the other args as well.